Sources/BraveNews/Composer/FeedDataSource.swift

```diff
@@ -768,11 +771,13 @@ public class FeedDataSource: ObservableObject {
   ) {
     // Ensure main thread since we're querying from CoreData
     dispatchPrecondition(condition: .onQueue(.main))
-    let lastVisitedDomains =
-      (try? History.suffix(200)
-        .lazy
-        .compactMap(\.url)
-        .compactMap { URL(string: $0)?.baseDomain }) ?? []
+    
+    var lastVisitedDomains: [String] = []
+    
+    historyAPI?.search(withQuery: "", maxCount: 200) { historyNodeList in
+      lastVisitedDomains = historyNodeList.compactMap { $0.url.baseDomain }
```

Comment:
Is this closure non-escaping? If not then this won't work and we'll need to wait until its done

Reply:
It is an nonnullable block on brave-core side that is translated into Swift as an @escaping so thanks for the catch. Fixed at https://github.com/brave/brave-ios/pull/7087/commits/7f0d662a53231d133f4d69b9fee05c97c988603a